robottelo/common/helpers.py

```diff
@@ -302,3 +302,34 @@ def patched_computeresource_init(self, server_config=None, **kwargs):
             computeresource_init(self, server_config, **kwargs)
             self._fields['url'].default = docker_url
         entities.ComputeResource.__init__ = patched_computeresource_init
+
+
+def prepare_import_data(tar_path=None):
+    """Fetch and uncompress the CSV files from the source."""
+    tmpdir = '/tmp/'
+    # if path to tar file not specified, download the default one
+    if tar_path is None:
+        tar_remote_path = conf.properties['transitions.export_tar.url']
+        cmd = u'wget {0} -O - | tar -xzvC {1}'.format(tar_remote_path, tmpdir)
+    else:
+        cmd = u'tar -xzvf {0} -C {1}'.format(tar_path, tmpdir)
+
+    files = {}
+    for file in ssh.command(cmd).stdout:
+        if file.endswith('activation-keys.csv'):
+            files['activation-keys'] = tmpdir + file
+        elif file.endswith('channels.csv'):
```

Comment:
To be honest I am not sure that it is good idea to have so long if/elif statement. From my previous experience it is always better and recommended to use switch-case approach for things like this. Could someone confirm or ban that suggestion?